File: src/main/java/net/igneo/icv/enchantment/EnchantType.java
package net.igneo.icv.enchantment;

public enum EnchantType {
    WEAPON,
    HELMET,
    CHESTPLATE,
    LEGGINGS,
    BOOTS,
    BOW,
    CROSSBOW,
    SHIELD,
    TRIDENT;
    
    public static boolean applicableSlot(EnchantType type, int slot) {
        switch (type) {
            case BOOTS: {
                return slot == 0;
            }
            case LEGGINGS: {
                return slot == 1;
            }
            case CHESTPLATE: {
                return slot == 2;
            }
            case HELMET: {
                return slot == 3;
            }
            case WEAPON, TRIDENT: {
                return slot == 4 || slot == 5;
            }
        }
        return false;
    }
}

File: src/main/java/net/igneo/icv/enchantment/EnchantType.kt
package net.igneo.icv.enchantment

enum class EnchantType {
    WEAPON,
    HELMET,
    CHESTPLATE,
    LEGGINGS,
    BOOTS,
    BOW,
    CROSSBOW,
    SHIELD,
    TRIDENT;

    companion object {
        @Suppress("UNREACHABLE_CODE")
        @JvmStatic
        fun applicableSlot(type: EnchantType, slot: Int): Boolean {
            return when (type) {
                BOOTS -> slot == 0
                LEGGINGS -> slot == 1
                CHESTPLATE -> slot == 2
                HELMET -> slot == 3
                WEAPON, TRIDENT -> slot == 4 || slot == 5

                BOW -> TODO()
                CROSSBOW -> TODO()
                SHIELD -> TODO()
            }
            return false
        }
    }
}
